cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -1515,14 +1540,31 @@ public void getAvailableTopics(ChannelTopicsListener listener) {
         return topics;
     }
 
+    public void addSubscriptionTags(String[] tagIds) {
+        this.tagIdsToAdd = tagIds;
+        currentPositionOfTagToAdd = 0;
+        addSubscriptionTag(tagIds[currentPositionOfTagToAdd]);
+        this.setOnAddTagCompletedListener(new OnAddTagCompleted() {
+            @Override
+            public void onAddTagCompleted() {
+                if (currentPositionOfTagToAdd != tagIdsToAdd.length - 1) {
+                    currentPositionOfTagToAdd++;
+                    addSubscriptionTag(tagIdsToAdd[currentPositionOfTagToAdd]);
+                }
+            }
+        });
+    }
+
     public void addSubscriptionTag(String tagId) {
-        Log.d("CleverPush", "addSubscriptionTag: " + tagId);
         this.waitForTrackingConsent(() -> new Thread(() -> this.getSubscriptionId(subscriptionId -> {
             if (subscriptionId != null) {
                 Set<String> tags = this.getSubscriptionTags();
                 if (tags.contains(tagId)) {
+                    if(onAddTagCompleted !=  null)
+                        onAddTagCompleted.onAddTagCompleted();
```

Comment:
`onAddTagCompleted` should be a second parameter to the addSubscriptionTag method NOT a global variable.

We will have 2 methods

`public void addSubscriptionTag(String tagId) {`
(will call the below method with second parameter: `null` )

`public void addSubscriptionTag(String tagId, AddTagListener callback) {`